docs/src/proposals/append-vec-storage.md

```diff
@@ -0,0 +1,231 @@
++--------------------------------------+
+|         ...                          |
++--------------------------------------+
+| account pubkey N                     |
++--------------------------------------+
+```
+
+```
++-------------------------------------------------------------+
+| Account Pubkey (32 bytes each)                              |
++-------------------------------------------------------------+
+| pubkey          | 32 bytes                                  |
++-------------------------------------------------------------+
+```
+
+#### Owners Block
+The owners block stores the owner's information.  An owner entry can be accessed
+via `owner_block_offset` + 32 + `owner_local_id` * `OWNER_ENTRY_SIZE`.
+
+```
++--------------------------------------+
+| Hash (32 bytes for the entire block) |
++--------------------------------------+
+| owner 1                              |
++--------------------------------------+
+| owner 2                              |
++--------------------------------------+
+|         ...                          |
++--------------------------------------+
+| owner N                              |
++--------------------------------------+
+```
+
+```
++-------------------------------------------------------------+
+| Owner (40 bytes each)                                       |
++-------------------------------------------------------------+
+| pubkey          | 32 bytes                                  |
+| write version   |  8 bytes                                  |
++-------------------------------------------------------------+
+```
+
+#### Account Data Blocks
+
+
+```
++------------------------------------+
+| Data blocks hash | 32 bytes        |
++------------------------------------+
+| Account data 1                     |
++------------------------------------+
+| Account data 2                     |
++------------------------------------+
+| Account data 3                     |
++------------------------------------+
+|          ...                       |
++------------------------------------+
+| Account data N                     |
++------------------------------------+
+```
+
+```
++-------------------------------------------------------------+
+| Account data block (variable size)                          |
++-------------------------------------------------------------+
+| rent_epoch      |  0-8 bytes                                |
+|                 |  0 byte if the account is rent exempt     |
+| data            |  N bytes                                  |
++-------------------------------------------------------------+
+```
+
+#### Footer
+The footer offers a quick check to make sure that the file is not
+truncated.  It also includes the hash of the entire AppendVec
+for verification purpose.
+
+```
++------------------------------------+
+| AppendVec Hash | 32 bytes          |
++------------------------------------+
+| Magic number   |  8 bytes          |
```

Comment:
This footer (and the format) will make these append vecs not usable for the ancient append vec process at the moment.

Reply:
Thanks for pointing this out.  Let me think more about it for ancient AppendVecs.